tests/test_embed/CMakeLists.txt

```diff
@@ -43,3 +43,13 @@ endforeach()
 add_dependencies(cpptest external_module)
 
 add_dependencies(check cpptest)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND PYTHON_VERSION VERSION_LESS 3.0)
+  if(CMAKE_CXX_STANDARD VERSION_LESS 17)
+    target_compile_options(test_embed PUBLIC -Wno-deprecated-register)
+    target_compile_options(external_module PUBLIC -Wno-deprecated-register)
+  else()
+    target_compile_options(test_embed PUBLIC -Wno-register)
+    target_compile_options(external_module PUBLIC -Wno-register)
```

Comment:
What about just adding both flags, without checking the standard version?

Reply:
I believe you can get a warning/error for an unrecognized flag.

Reply:
Right. Clang warns about that, but gcc doesn't. Hah, we can add `-Wno-register`, `-Wno-deprecated-register` *and* `-Wno-unknown-warning-option`. Just kidding, of course.

Reply:
Wait, gcc doesn't know `-Wno-register`? I do build with that.

Reply:
I was talking about warning on, literally, `-Wno-whatever`. `man gcc` does mention `-Wregister` so you're fine.

Reply:
Alright! Follow-up question, then: why is there a test for `CMAKE_CXX_COMPILER_ID MATCHES "Clang"` when this could also be gcc? Or does gcc not complain about `register`; I thought it did?

Reply:
```       -Wregister (C++ and Objective-C++ only)
           Warn on uses of the "register" storage class specifier, except when it is part of the GNU Explicit Register Variables extension.  The use of the
           "register" keyword as storage class specifier has been deprecated in C++11 and removed in C++17.  Enabled by default with -std=c++17.
```

So the warning will be turned on by default when a user switches on C++17. GCC 11 will default to C++17.

Reply:
~~Does this show up though? We are compiling with GCC + C++17 with warnings as errors.~~ I bet we are not setting ` CMAKE_CXX_EXTENSIONS=OFF`, actually. I'd rather leave it as is (not any worse than before) then tackle this during the CMake overhaul.

Reply:
OK!